Tasks/AzureMysqlDeployment/operations/FirewallOperations.ts

```diff
@@ -0,0 +1,124 @@
+     * @param firewallRule        firewallRule i.e name and Ip Adress range
+     * @param resourceGroupName   mysql server resource group name
+     * 
+     * @returns operation is success or failure
+     */
+    public async addFirewallRule(serverName: string, firewallRule: FirewallRule, resourceGroupName: string): Promise<void> {
+        let defer = Q.defer<void>();
+        this._azureMysqManagementClient.firewallRules.createOrUpdate(resourceGroupName, serverName, firewallRule.getName(), firewallRule, (error, result, request, response) => {
+            if(error){
+                task.debug("Getting error during adding firewall rule: "+ error);
+                defer.reject(new Error(task.loc("NotAbleToAddFirewallRule", error)));
+            }else{
+                defer.resolve();
+            }
+        });
+        return defer.promise;
+    }
+
+    /**
+     * Delete firewall rule for mysql server 
+     * @param serverName           mysql server name
+     * @param resourceGroupName    mysql server resource group name
+     * 
+     * @returns operation is success or failure
+     */
+    public async deleteFirewallRule(serverName: string, resourceGroupName: string): Promise<void> {
+        let defer = Q.defer<void>();
+        this._azureMysqManagementClient.firewallRules.delete(resourceGroupName, serverName, this._firewallName, (error, result, request, response) => {
+            if(error){
+                task.debug("Getting error during deleting firewall rule: "+ error);
+                defer.reject(new Error(task.loc("NotAbleToDeleteFirewallRule", error)));
+            }else{
+                defer.resolve();
+            }
+        });
+        return defer.promise;
+    }
+
+    /**
+     * To check agent box has permission to connect with mysqlServer or not. If not then add firewall rule to whitelist this IP.
+     * @param azureMysqlTaskParameter    task input parameters
+     * @param sqlClient                  mysql client
+     * @param resourceGroupName          mysql server resource group name
+     * 
+     * @returns                          firewall rule added or not 
+     */
+    public async invokeFirewallOperations(azureMysqlTaskParameter: AzureMysqlTaskParameter, sqlClient: ISqlClient, resourceGroupName: string) : Promise<boolean> {
+        var defer = Q.defer<boolean>();
+        if(azureMysqlTaskParameter.getIpDetectionMethod() ==='IPAddressRange'){
+            this._preparefirewallRule(azureMysqlTaskParameter.getServerName(), azureMysqlTaskParameter.getStartIpAddress(), azureMysqlTaskParameter.getEndIpAddress(), resourceGroupName, "IPAddressRange_" + this._getFirewallRuleName()).then(() =>{
+                let firewallConfiguration: FirewallConfiguration = sqlClient.getFirewallConfiguration();
+                console.log(" firewall conf " +JSON.stringify(firewallConfiguration));
+                if(!firewallConfiguration.isIpAdressAlreadyAdded()){
```

Comment:
Should there be some empty lines for better clarity?

Reply:
Done